src/tracer.js

```diff
@@ -228,6 +228,7 @@ export default class Tracer {
             ctx.parentId = null;
             ctx.flags = flags;
         } else {
+            this.processDeferredSampling(parent, operationName, internalTags);
```

Comment:
Since we narrowed the scope, would't the `parent` in this case always be something we just serialized from the wire? The _only_ child is going to be the server-side span.

And as a more general principle, if the parent wants to ensure that its sampling decision is fixed and should affect all children, then it's he parent who should make the change, not as a side effect of creating a child.